pageserver/src/tenant/timeline.rs

```diff
@@ -4107,7 +4131,15 @@ impl Timeline {
                 continue;
             }
 
-            let last_activity_ts = l.access_stats().latest_activity();
+            let last_activity_ts = match l.access_stats().latest_activity() {
+                Some(ts) => ts,
+                None => {
+                    // If we're under disk pressure, we should not hide this layer's existence.
+                    // Log a warning and
+                    warn!(layer=%l.filename().file_name(), "latest activity not available, did we forget to add a residence event?");
```

Comment:
Re: `%l.filename().file_name()`:

I think we should try to refactor a `Display` impl to Layer in addition to the verbose `Debug` and move towards using just these. I am pretty sure we discussed this recently and I may have gone with `layer_id` then. Or maybe it was for the absolute path logging issue, to use layerid instead of file name.

Reply:
Agree, separate issue though. Can you create one?

Reply:
Created https://github.com/neondatabase/neon/issues/4145